torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2939,6 +2956,21 @@ std::tuple<Tensor, Tensor> triangular_solve_backward(
   return std::tuple<Tensor, Tensor>{grad_b, grad_a};
 }
 
+Tensor triangular_solve_jvp(
+  const Tensor& X, const Tensor& A,
+  const Tensor& dA, const Tensor& dB,
+  const bool upper,
+  const bool transpose,
+  const bool unitriangular
+) {
+  return generic_solve_jvp(
+    [=](const Tensor& A, const Tensor& B) {
```

Comment:
```suggestion
    [upper, transpose, unitriangular](const Tensor& A, const Tensor& B) {
```
Otherwise it might need to copy all the tensors (I don't know if the compiler is allowed to ellide the copies).
Same below.

Reply:
Not sure it will happen either way since afaik Tensors are not being copied on `operator=`.

Reply:
It was more about the copies of tensors and the reference bumps. A very very minor thing really.

Reply:
You can also capture everything by reference because the lambda never outlives the current scope?